### PR TITLE
Implement optional prefix for commands

### LINKED ISSUE #271: Implement optional prefix
Make prefix optional for commands listed below:
- [ ] open
- [ ] edit
- [ ] delete

### PR DESCRIPTION
**Pull request by ljiazh3ng**
resolves #271 
This PR makes prefix optional for commands listed below:
- [x] open
- [x] edit
- [x] delete

### CODE REVIEW

**Review by firzanarmani — APPROVED** [review]
LGTM, don't forget to update to latest master first! 😄

**Review by firzanarmani — APPROVED** [review]

File: src/main/java/com/notably/logic/parser/DeleteCommandParser.java [hunk 1]
```diff
@@ -33,12 +33,14 @@ public DeleteCommandParser(Model notablyModel) {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_TITLE);
 
+        String title;
         if (!ParserUtil.arePrefixesPresent(argMultimap, PREFIX_TITLE)
                 || !argMultimap.getPreamble().isEmpty()) {
-            throw new ParseException(String.format("Invalid Command"));
+            title = args.trim();
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/parser/DeleteCommandParser.java:39]: Let's check for an empty title. That is, throw a `ParseException` if the user types `delete` only, without any arguments!

File: src/main/java/com/notably/logic/parser/EditCommandParser.java [hunk 2]
```diff
@@ -24,13 +24,14 @@
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_BODY);
 
+        String body;
         if (!ParserUtil.arePrefixesPresent(argMultimap, PREFIX_BODY)
                 || !argMultimap.getPreamble().isEmpty()) {
-            throw new ParseException(String.format("Invalid Command"));
+            body = args.trim();
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/parser/EditCommandParser.java:30]: Let's check for an empty title. That is, throw a `ParseException` if the user types `edit` only, without any arguments!

File: src/main/java/com/notably/logic/parser/OpenCommandParser.java [hunk 3]
```diff
@@ -34,12 +34,14 @@ public OpenCommandParser(Model notablyModel) {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_TITLE);
 
+        String titles;
         if (!ParserUtil.arePrefixesPresent(argMultimap, PREFIX_TITLE)
                 || !argMultimap.getPreamble().isEmpty()) {
-            throw new ParseException(String.format("Invalid input"));
+            titles = args.trim();
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/parser/OpenCommandParser.java:40]: Let's check for an empty title. That is, throw a `ParseException` if the user types `open` only, without any arguments!

File: src/test/java/com/notably/logic/parser/NotablyParserTest.java [hunk 4]
```diff
@@ -94,13 +95,45 @@ public void parseCommand_openCommandInput_openCommand() throws Exception {
...
+
+        assertTrue(command instanceof EditCommand);
+    }
+
+    @Test
+    public void parseCommand_editCommandInputNoPrefix_editCommand() throws Exception {
+        Command command = parser.parseCommand("edit ../CS2103").get(0);
+
+        assertTrue(command instanceof EditCommand);
+    }
```
Comment (kevinputera) [inline src/test/java/com/notably/logic/parser/NotablyParserTest.java]: Edit's `-b` flag should indicate body right? Could this be a typo?

**Review by kevinputera — CHANGES_REQUESTED** [review]
Let's improve our error handling!

File: src/main/java/com/notably/logic/parser/DeleteCommandParser.java (hunk 1 above)
Comment (ljiazh3ng) [reply src/main/java/com/notably/logic/parser/DeleteCommandParser.java:39]: Done!

File: src/main/java/com/notably/logic/parser/EditCommandParser.java (hunk 2 above)
Comment (ljiazh3ng) [reply src/main/java/com/notably/logic/parser/EditCommandParser.java:30]: I think this is ok, lets give the user to option to make the body empty.

File: src/main/java/com/notably/logic/parser/OpenCommandParser.java (hunk 3 above)
Comment (ljiazh3ng) [reply src/main/java/com/notably/logic/parser/OpenCommandParser.java:40]: done!

File: src/test/java/com/notably/logic/parser/NotablyParserTest.java (hunk 4 above)
Comment (ljiazh3ng) [reply src/test/java/com/notably/logic/parser/NotablyParserTest.java]: Oh yes, i've made the necessary changes!

File: src/main/java/com/notably/logic/parser/DeleteCommandParser.java [hunk 5]
```diff
@@ -33,12 +33,17 @@ public DeleteCommandParser(Model notablyModel) {
...
+            if (title.equals("")) {
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/parser/DeleteCommandParser.java]: You should actually do `title.isEmpty()`, more idiomatic!

File: src/main/java/com/notably/logic/parser/OpenCommandParser.java [hunk 6]
```diff
@@ -34,12 +34,17 @@ public OpenCommandParser(Model notablyModel) {
...
+            if (titles.equals("")) {
+                throw new ParseException("Path cannot be empty");
+            }
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/parser/OpenCommandParser.java]: You should actually do `title.isEmpty()`, more idiomatic!

**Review by kevinputera — APPROVED** [review]
LGTM, some minor comments below!

File: src/main/java/com/notably/logic/parser/DeleteCommandParser.java (hunk 5 above)
Comment (ljiazh3ng) [reply src/main/java/com/notably/logic/parser/DeleteCommandParser.java]: will do

File: src/main/java/com/notably/logic/parser/OpenCommandParser.java (hunk 6 above)
Comment (ljiazh3ng) [reply src/main/java/com/notably/logic/parser/OpenCommandParser.java]: will do

**Pull request merged**